src/secret_sharing/replicated/malicious/additive_share.rs

```diff
@@ -23,16 +23,37 @@ use std::{
 use typenum::Unsigned;
 
 #[derive(Clone, PartialEq, Eq)]
-pub struct AdditiveShare<V: SharedValue> {
+pub struct AdditiveShare<V: SharedValue + ExtendableField> {
     x: SemiHonestAdditiveShare<V>,
-    rx: SemiHonestAdditiveShare<V>,
+    rx: SemiHonestAdditiveShare<V::LargeFieldType>,
 }
 
-impl<V: SharedValue> SecretSharing<V> for AdditiveShare<V> {
+pub trait ExtendableField {
+    type LargeFieldType: Field;
+    fn get_induced_value(&self) -> Self::LargeFieldType;
+}
+
+impl ExtendableField for Fp32BitPrime {
+    type LargeFieldType = Fp32BitPrime;
+
+    fn get_induced_value(&self) -> Self::LargeFieldType {
+        *self
+    }
+}
+
+impl ExtendableField for Gf2 {
+    type LargeFieldType = Gf32Bit;
+
+    fn get_induced_value(&self) -> Self::LargeFieldType {
+        Gf32Bit::try_from(self.as_u128()).unwrap()
```

Comment:
@taikiy - is there a better way to do this?

Reply:
I think the current implementation is good. Maybe just move `ExtendableField` to `field.rs`? Also, line 48 can be

`Self::LargeFieldType::try_from(self.as_u128()).unwrap()`

Reply:
Don't we have a truncate version as well?

Reply:
Since this is specific to the GF2 => Gf32Bit transition, I was thinking we could literally just clone the least significant byte of storage. How can I do that @taikiy ?

Reply:
This code does exactly what you described - in a safe way via `try_from`. `truncate_from`, which is called by `try_from`, copies the least significant bits and creates a new field instance.